### ISSUE TITLE
url_preview media cache is not expired when background database migrations are running

### ISSUE THREAD

**Issue by richvdh** [matrix-org/synapse#13637]
Normally, any media downloaded as part of a url preview is deleted after a day. However, while a background database migration is running, the process is stopped. If there is a long-running database migration, this can therefore cause the media store to fill up with old preview files.

This logic was added in https://github.com/matrix-org/synapse/pull/2697 to make sure that we didn't try to run the expiry without an index on `local_media_repository.created_ts`; the original logic that needs that index was added in https://github.com/matrix-org/synapse/pull/2478 (in [`get_url_cache_media_before`](https://github.com/matrix-org/synapse/pull/2478/files#diff-5772984c46f21c905ff00a9b6aebcb39a5fa21fec74c046fbd968b4721616a16R268-R273), as amended by https://github.com/matrix-org/synapse/commit/93247a424a5068b088567fa98b6990e47608b7cb), and is still present.

It looks like too broad a constraint: instead, we should use `has_completed_background_update` to check if the *specific* background update we care about has completed. However, given you'd have to be upgrading from before Synapse v1.0.0 (which created the `v54` schema snapshot) to *not* have this index in place... we should probably just drop the check altogether.

**Comment by richvdh** [comment]
TL;DR: let's just remove [this check](https://github.com/matrix-org/synapse/blob/v1.65.0/synapse/rest/media/v1/preview_url_resource.py#L735).

**Comment by clokep** [comment]
> any media downloaded as part of a url preview is deleted after a day

Just to avoid confusion -- they're deleted after [two days](https://github.com/matrix-org/synapse/blob/1381563988c6dc7a2b8801b736b1f0c663970da8/synapse/rest/media/v1/preview_url_resource.py#L67). I don't think this changes any of the rest of the issue though. Thanks for researching!
